tests/unit/api/c/receiver/test_receiver.cpp

```diff
@@ -66,6 +67,109 @@ class TestReceiver : public ::testing::Test
     sacn_receiver_deinit();
     sacn_mem_deinit();
   }
+
+  void TestChangeUniverse(sacn_ip_support_t ip_supported)
```

Comment:
This is kind of a long function - one idea that might make this test code easier to understand, is if you put all the common setup code at the beginning of the function in its own "Setup" function, e.g. SetupUniverseChangeTest(). Then transfer the code in the big switch statement below to each individual TEST_F() function. Just a thought.